### PR TITLE
Add API method for editing temp of "Current Extruder" on a multiple Extruder printer.

### PR DESCRIPTION
**Pull request by ccatlett1984**
This solves  :
https://github.com/OctoPrint/OctoPrint/issues/5018
Adds a "Current_Extruder" option for the toolNum element for setting hotend temp via a REST call.
when the new element is used, it leverages the "current_extruder" variable in the gcode call, since we currently have no tracking of which toolhead/extruder is active and there is no call to retrieve the data from the printer.

  * [x] Your changes are not possible to do through a plugin and relevant
    to a large audience (ideally all users of OctoPrint)
  * [x] If your changes are large or otherwise disruptive: You have
    made sure your changes don't interfere with current development by
    talking it through with the maintainers, e.g. through a
    Brainstorming ticket
  * [x] Your PR targets OctoPrint's `devel` branch if it's a completely
    new feature, or `maintenance` if it's a bug fix or improvement of
    existing functionality for the current stable version (no PRs
    against `master` or anything else please)
  * [x] Your PR was opened from a custom branch on your repository
    (no PRs from your version of `master`, `maintenance`, or `devel`
    please), e.g. `dev/my_new_feature` or `fix/my_bugfix`
  * [x] Your PR only contains relevant changes: no unrelated files,
    no dead code, ideally only one commit - rebase and squash your PR
    if necessary!
  * [x] Your changes follow the existing coding style
  * [x] If your changes include style sheets: You have modified the
    `.less` source files, not the `.css` files (those are generated
    with `lessc`)
  * [x] You have tested your changes (please state how!) - ideally you
    have added unit tests
  * [x] You have run the existing unit tests against your changes and
    nothing broke
  * [x] You have added yourself to the `AUTHORS.md` file :)

#### What does this PR do and why is it necessary?
This adds an additional endpoint for controlling the "current" toolhead on a multiple toolhead printer, leveraging the "current_extruder" gcode variable, since we have no tracking of which toolhead is active.

#### How was it tested? How can it be tested by the reviewer?
Have at least 2 toolheads, select toolhead two (T1).
Make the following POST request:
POST /api/printer/tool HTTP/1.1
Content-Type: application/json
{
  "command": "target",
  "targets": {
    "toolCurrent_Extruder": 220
  }
}

The terminal should show the following being sent:
M104 T{current_extruder} S{220}

#### What are the relevant tickets if any?
https://github.com/OctoPrint/OctoPrint/issues/5018

### CODE REVIEW

**Comment by foosel** [comment]
I've changed the implementation a bit.

First of all you forgot to adjust the regexes used for validating tool names. Then, I really wasn't happy with something like `toolCurrent_Extruder` - instead it's now just `tool` without a number, and it is only supported for setting a temperature, which is the only place where it even makes sense (so I also added another validation regex to ensure that on e.g. setting an offset or selecting a tool it would not be supported). 

Then I also adjusted the actual REST API endpoint, which also contains a sanity check which got forgotten, fixed a bug in the virtual printer plugin (`M104 S120` would set the temperature of the first tool, instead of the current one) and finally updated the docs as well.

So, long story short, something like this works now:

```
curl --header "Authorization: Bearer ..." http://example.com/api/printer/tool --json '{"command": "target", "targets": {"tool": 120}}'
```

and is also documented correctly.
